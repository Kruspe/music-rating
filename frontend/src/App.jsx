import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import './App.css';
import { Auth0Provider } from '@auth0/auth0-react';
import { createTheme, ThemeProvider } from '@mui/material';
import Content from './Content';

const queryClient = new QueryClient();

const darkTheme = createTheme({
  palette: {
    mode: 'dark',
  },
});

function App() {
  return (
    <Auth0Provider
      domain="https://musicrating.eu.auth0.com"
      clientId={process.env.REACT_APP_CLIENT_ID}
      redirectUri={process.env.REACT_APP_DOMAIN_NAME.includes('localhost')
        ? process.env.REACT_APP_DOMAIN_NAME : `https://${process.env.REACT_APP_DOMAIN_NAME}`}
      audience={process.env.REACT_APP_DOMAIN_NAME.includes('localhost')
        ? process.env.REACT_APP_DOMAIN_NAME : `https://api.${process.env.REACT_APP_DOMAIN_NAME}`}
    >
      <ThemeProvider theme={darkTheme}>
        <QueryClientProvider client={queryClient}>
          <Content />
        </QueryClientProvider>
      </ThemeProvider>
    </Auth0Provider>
  );
}

export default App;
