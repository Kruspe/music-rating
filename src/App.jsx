import Amplify from 'aws-amplify';

import awsExports from './aws-exports';
import './App.css';

Amplify.configure(awsExports);

const App = () => (<div>Hello World</div>);

export default App;
