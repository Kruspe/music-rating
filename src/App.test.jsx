import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';

jest.mock('./tabs/TabBar', () => (() => (<p>TabBar</p>)));

describe('App', () => {
  it('should render no content when not signedIn', () => {
    const { container } = render(<App authState="signIn" />);
    expect(container).toBeEmptyDOMElement();
  });

  it('should render TabBar when signedIn', async () => {
    render(<App authState="signedIn" />);
    expect(await screen.findByText('TabBar')).toBeVisible();
  });
});
