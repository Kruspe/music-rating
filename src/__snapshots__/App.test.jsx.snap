// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should renders content correctly 1`] = `<Rating />`;
